README.rst

```diff
@@ -0,0 +1,329 @@
+and overrides the ``configure()`` method to define all of the application components:
+
+.. code:: java
+
+  public class PurchaseWorkflowApp extends AbstractApplication {
+    @Override
+    public void configure() {
+      setName("PurchaseWorkflowApp");
+      setDescription("Application describing the Workflow");
+
+      addStream(new Stream("purchaseEvents"));
+
+      addMapReduce(new PurchaseEventParser());
+      addMapReduce(new PurchaseCounterByUser());
+      addMapReduce(new PurchaseCounterByProduct());
+      addWorkflow(new PurchaseWorkflow());
+
+      scheduleWorkflow(Schedules.createTimeSchedule("HourlySchedule", "Schedule execution every 1 hour", "0 * * * *"),
+                       "PurchaseWorkflow");
+
+      addService(new PurchaseResultService());
+
+      createDataset("purchaseRecords", KeyValueTable.class);
+      createDataset("userPurchases", KeyValueTable.class);
+      createDataset("productPurchases", KeyValueTable.class);
+    }
+  }
+  
+The ``PurchaseWorkflowApp`` application defines a new `Stream
+<http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/streams.html>`__
+where purchase events are ingested. Once the data is
+ingested, the events can be processed in real time or batch. In our
+application, we will process the events in batch using the
+``PurchaseWorkflow`` program and compute the total purchases made by a particular user
+and the total purchases made for a particular product in a specific time range. We will use three MapReduce
+programs ``PurchaseEventParser``, ``PurchaseCounterByUser``, and ``PurchaseCounterByProduct`` to apply
+different processing on the purchase events and the Workflow ``PurchaseWorkflow`` to connect these MapReduce
+programs in a desired fashion.
+
+The result of the Workflow execution is persisted into a Dataset; the
+application uses the ``createDataset`` method to define the Dataset. We use three datasets:
+``purchaseRecords`` to store the valid parsed purchase events; ``userPurchases`` to store the total purchases
+made by a particular user; and ``productPurchases`` to store the total purchases made for each product.
+The ``Purchase`` class defines the type used to store the parsed purchase events.
+
+The application also adds a custom Workflow action ``EmailNotifier``. When a Workflow executes a custom action,
+it invokes the ``run`` method in the action. In ``EmailNotifier``, we only add a log statement; however it could be
+customized to send emails to the concerned parties.
+
+The ``PurchaseWorkflow`` is scheduled to execute every hour.
+
+Finally, the application adds a service for querying the results from the Dataset.
+
+Let's take a closer look at the Workflow program.
+
+The ``PurchaseWorkflow`` extends an `AbstractWorkflow
+<http://docs.cdap.io/cdap/current/en/reference-manual/javadocs/co/cask/cdap/api/workflow/AbstractWorkflow.html>`__
+class and overrides the ``configure()`` method:
+
```

Comment:
Not sure what @awholegunch thinks about this, but I think we normally do literal include for code extracts. Or does that not work in guides?